### DISCUSSION TITLE
Adding a Cart Value / Order Price Condition for Discounts

### DISCUSSION THREAD

**Discussion by bqst** [medusajs/medusa#5642] [Feature Requests]
I would like to suggest the addition of a condition related to the total cart / order price value in the discount creation process. The idea is to enable the discount to apply only if the total amount in the shopping cart reaches a certain threshold. This functionality would allow us to create more targeted and strategic promotions.

We could extend the DiscountCondition entity with something like :
```
export interface DiscountCondition {
  type:
        // ...
    | "cart"

  /**
   * The operator for the condition, used to compare cart total with the specified amount.
   * For example, 'gt' means the cart total should be greater than the specified amount.
   */
  operator?: "gt" | "lt" | "eq" | "gte" | "lte"

  /**
   * Optional: Specific cart associated with this condition, if `type` is `cart`.
   */
  cart?: Cart

  /**
   * Monetary amount used for comparison with the cart total, relevant if `type` is `cart`.
   */
  amount?: number

  // ... (rest of the interface)
}
```

**Comment by srindom** [comment]
Thank you for your input. This is a great idea. Do you have other thoughts on how the discounts and promotions module could be made better?

**Reply by bqst** [reply]
Yep, here are a few additional ideas that could further improve the discounts:

- **Tiered Discounts**: Implementing tiered discounts based on cart value thresholds, for example, customers could receive a 10% discount if their cart total exceeds 100€, 15% for 200€, and so on.
- **Bundling Products**: Encourage customers to purchase complementary products together by offering discounts on bundles. For example, buying a pair of shoes and a matching bag together might trigger a special discount. 
- **Multi-buy Discounts**: Discounts for purchasing multiple quantities of the same item. For example, "Buy 2, Get 1 Free" offers.

I've used Shopify quite a bit, and it would be great to be able to approach the mode of creating conditions similar to this: https://help.shopify.com/en/manual/discounts/discount-types/percentage-fixed-amount

**Comment by jstmrby** [comment]
Any plans on implementing it?
I find these type of promotions really popular. It would be nice to support them.
